src/main/perl/Path.pm

```diff
@@ -544,6 +564,107 @@ sub directory
 }
 
 
+=item _make_link
+
+This method is mainly a wrapper over C<LC::Check::link>
+returning the standard CAF::Path return values. Every option
+supported by C<LC::Check::link> is supported. C<NoActionFlag>
+is handled by C<LC::Check::link>.
+
+This in internal method, not supposed to be called directly.
+Either call C<symlink> or C<hardlink> public methods instead.
+
+=cut
+
+sub _make_link
+{
+    my ($self, $target, $link_path, %opts) = @_;
+    my $link_type = $opts{hard} ? "hardlink" : "symlink";
+
+    $link_path = $self->_untaint_path($link_path, $link_type) || return;
+    $target = $self->_untaint_path($target, $link_type) || return;
+
+    $self->_reset_exception_fail($link_type);
+
+    my $status = $self->_safe_eval(\&LC::Check::link,
```

Comment:
Because I missed it! You mentioned it earlier pointing at `directory()` and this is where I found `_safe_eval()`. I thought it was what you were refering too. I'll double check if it makes things easier.